### PR TITLE
Switch to using New Tables

### PR DESCRIPTION
**Pull request by JDaniel41**
DO NOT MERGE UNTIL #100 is merged

This change has the lambdas begin writing to the new tables, while also continuing to write to the old table. The lambdas are also able to read from the old tables.

### CODE REVIEW

**Comment by JDaniel41** [comment]
So, I'm not sure if we need to have #93 merged before we merge this. Currently, it continues reading from the old table, and using that as our "ground truth" (without consulting the new tables). If we run with the PR as is, I feel like we can go ahead and merge this and run the migration after.  Thoughts? @ophilli @dggorsen

File: cdk/database/__init__.py
Comment (dggorsen) [inline cdk/database/__init__.py]: I think there is value in keeping this commented incase we arent able to clean up this code by the end of the semester

File: cdk/visit/lambda_code/log_visit/log_visit.py
Comment (dggorsen) [inline cdk/visit/lambda_code/log_visit/log_visit.py:58]: For all reads- we should be reading from the old table, right?

File: cdk/visit/lambda_code/log_visit/log_visit.py
Comment (dggorsen) [inline cdk/visit/lambda_code/log_visit/log_visit.py]: In a perfect world, we would read from both tables
1. Read from old table (primary)
2. Read from new table (secondary)
3. Equality comparison between the two. If new table has no data or stale data, write the item from the old table to the new table
4. Return data from the first read (primary table)

In our case, that is probably overkill. We should only need to read from the old table and write to both. We will handle backfilling from a script

File: cdk/visit/lambda_code/log_visit/log_visit.py
Comment (dggorsen) [inline cdk/visit/lambda_code/log_visit/log_visit.py]: yep! Effectively we want **everything** to behave exactly as it was before. We just want to write items to the new table as well

**Review by dggorsen — CHANGES_REQUESTED** [review]

File: cdk/visit/lambda_code/log_visit/log_visit.py
Comment (JDaniel41) [reply cdk/visit/lambda_code/log_visit/log_visit.py:58]: Yep! Thanks for the catch!

File: cdk/visit/lambda_code/log_visit/log_visit.py
Comment (dggorsen) [inline cdk/visit/lambda_code/log_visit/log_visit.py:63]: non blocking but why are we returning the count here?

File: cdk/visit/lambda_code/register_user/register_user.py
Comment (dggorsen) [inline cdk/visit/lambda_code/register_user/register_user.py]: Does minor exist yet?

Comment (dggorsen) [inline cdk/visit/lambda_code/register_user/register_user.py]: We have an awesome opportunity to change our item structure here, and I suggest we take advantage of that while we can. Couple suggestions:

1. Lets make sure that we have a standard format for key casing. We have a mixture of camelCase, PascalCase, and snake_case. Any of the above is fine but we should pick a single standard.
2. Id suggest changing "major" and "minor" to their plural forms since they will be arrays. 
3. we might want to change grad_date to grad_semester, and we will want to do that conversion for whatever the frontend sends to the backend here. e.g. if the FE still sends a date, we should convert the date to a semester


Lastly, I would suggest that we use dynamodb json rather than standard json for putting our items here, because it will allow us to create strongly typed data. This will come in handy for serializing things like arrays and string sets. See https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.put_item for more details

**Review by dggorsen — CHANGES_REQUESTED** [review]

File: cdk/visit/lambda_code/log_visit/log_visit.py
Comment (JDaniel41) [reply cdk/visit/lambda_code/log_visit/log_visit.py:63]: This is used to check and see if the user has registered before. In the original table, they would be listed as the `PK` if they have registered.

File: cdk/visit/lambda_code/register_user/register_user.py
Comment (JDaniel41) [reply cdk/visit/lambda_code/register_user/register_user.py]: Yep!

**Review by dggorsen — APPROVED** [review]
Really great work on this! Let me know when to merge

**Pull request merged**
